### ISSUE TITLE
Upgrade to Rails 7.1

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#11673]
### What we should change and why (this is tech debt)

[The next Rails version has been released.](https://rubyonrails.org/2023/10/5/Rails-7-1-0-has-been-released) There are some performance optimisations which would be nice to have and some new tools we can explore. But there are several steps we need to take first:

- [ ] Wait for all our dependencies to be compatible with Rails 7.1.
- [ ] #7853
- [ ] Address any other existing deprecation warnings.
- [ ] Follow the [upgrade guide](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-7-0-to-rails-7-1).

### Impact and timeline

Not critical at the moment. But good to chip away at this issue.

**Comment by dacook** [comment]
Just noting that it would be worth waiting until the Ubuntu server upgrades](openfoodfoundation/ofn-install#157) are done, then we can upgrade Node, then Rails.

**Comment by dacook** [comment]
I just discovered that our current Node version is blocked until we get all servers onto Ubuntu 20 🤦 

But the node version doesn't seem to be a blocker for the Rails upgrade. At least, I didn't find any evidence online.
In a previous CI run (https://github.com/openfoodfoundation/openfoodnetwork/pull/12492), many of the specs succeeded, so I guess the current node version is fine.

**Comment by dacook** [comment]
I think all dependencies must be ready for Rails 7.1, because at least some CI runs succeed: https://github.com/dacook/openfoodnetwork/actions/runs/13106684816

**Comment by sigmundpetersen** [comment]
Hey @piyush828-design 👋 
This is the issue for Rails 7.1 if you wanted to have a look

**Comment by piyush828-design** [comment]
Hey @sigmundpetersen , thanks for navigating me to here. Here, I'm bit confused. Are we done with the upgrade to rails 7.1?
Or if any issue we are facing while upgrading ?

**Comment by sigmundpetersen** [comment]
You can see in the gemfile that we are on 7.0.8
https://github.com/openfoodfoundation/openfoodnetwork/blob/e900c95213df453ea127a81cef76996adfbeafaa/Gemfile.lock#L556 

I guess first step is to review the current build of main branch to see if there are any remaining Deprecation warnings to tackle before moving towards 7.1.

After that you can submit a PR bumping to 7.1 in the gemfile and observe what kind of build errors we get.
And at the same time following the upgrade guide as mentioned in the issue description.

@openfoodfoundation/reviewers would have further comments on the rails upgrade process

**Comment by dacook** [comment]
Yes I would start with the issue description. I'm aware of one remaining deprecation warning, which I just added to the description above. So it would be great if you could start with that one please.

**Comment by piyush828-design** [comment]
Hey @dacook, are you talking about https://github.com/dacook/openfoodnetwork/actions/runs/13106684816  ?

**Comment by dacook** [comment]
Regarding the deprecation warning, I meant: https://github.com/openfoodfoundation/openfoodnetwork/issues/13270

**Comment by sigmundpetersen** [comment]
Hey @dacook I wonder if we found another deprecation https://app.bugsnag.com/yaycode/open-food-network-rails/errors/6811fe8bf0d0ce3f5d919960?filters[error.status]=open&filters[event.since]=30d

Could you have a look and perhaps create an issue?

**Comment by sigmundpetersen** [comment]
Just noting that there is an ongoing draft PR in #13232 
You could either coordinate the efforts or maybe you want to await a bit @piyush828-design . Up to you 🙂

**Comment by chitty** [comment]
Thanks @sigmundpetersen, I looked into this a while ago and just had some cycles now to continue. Just got the Rails 7.1 branch passing CI 🎉 

To make the PR easier to review and less risky to ship, I'm thinking about splitting some of the commits into their own PRs. Mainly https://github.com/openfoodfoundation/openfoodnetwork/pull/13232/commits/2b9289e967a41899556128fbd867f155cde140a5.

Other than that, there's the deprecations mentioned in this thread. I have not looked into those, do you want to work on them @piyush828-design?

**Comment by dacook** [comment]
> I wonder if we found another deprecation [https://app.bugsnag.com/yaycode/open-food-network-rails/errors/6811fe8bf0d0ce3f5d919960?filters[error.status]=open&filters[event.since]=30d](https://app.bugsnag.com/yaycode/open-food-network-rails/errors/6811fe8bf0d0ce3f5d919960?filters%5Berror.status%5D=open&filters%5Bevent.since%5D=30d)

Thanks Sigmund, yes it looks like we need to address this before the upgrade, so I've added it as a sub-issue.

**Comment by sigmundpetersen** [comment]
We are on 7.1 so I think we can close this and move to #13290 
Should we move those 2 sub-issues to the 7.2 issue?

**Comment by mkllnk** [comment]
Sub-issues moved.
